hypervisor/arch/x86/vtd.c

```diff
@@ -1284,22 +1285,34 @@ void resume_iommu(void)
 
 void init_iommu(void)
 {
-	uint16_t bus;
-	uint16_t devfun;
-
 	INIT_LIST_HEAD(&dmar_drhd_units);
 	INIT_LIST_HEAD(&iommu_domains);
 
 	spinlock_init(&domain_lock);
 
 	register_hrhd_units();
+}
+
+void init_iommu_vm0_domain(struct vm *vm0)
+{
+	uint16_t bus;
+	uint16_t devfun;
+
+	vm0->iommu = create_iommu_domain(vm0->vm_id,
+		HVA2HPA(vm0->arch_vm.nworld_eptp), 48U);
 
-	host_domain = create_host_domain();
+	vm0_domain = (struct iommu_domain *) vm0->iommu;
 
 	for (bus = 0U; bus <= IOMMU_INIT_BUS_LIMIT; bus++) {
 		for (devfun = 0U; devfun <= 255U; devfun++) {
-			add_iommu_device(host_domain, 0U,
-					 (uint8_t)bus, (uint8_t)devfun);
+#ifdef HV_DEBUG
+			if ((bus == platform_uart_bus) &&
```

Comment:
Having a 2nd thinking, we are setting up IOMMU tables for all possible BDFs...  This is not wrong, but redundant already. Why we care about adding one more entry for UART BDF?

I thought if we skip this, we can simplify the code.

Reply:
Since Hypervisor doesn't do UART DMA, it is safe to skip it.